rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+        `GET /namespaces?parent=a` and must return a namespace, ["a", "b"].
+      operationId: listNamespaces
+      parameters:
+        - $ref: '#/components/parameters/parent'
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  namespaces:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Namespace'
+        401:
+          description: Unauthorized
+        404:
+          $ref: '#/components/responses/404Response'
+
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description:
+        Create a namespace, with an optional set of properties.
+        The server might also add properties, such as last_modified_time etc.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        200:
+          description: OK
+        401:
+          description: Unauthorized
+        409:
+          description: Conflict - The namespace already exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NamespaceAlreadyExists:
+                  $ref: '#/components/examples/NamespaceAlreadyExistsError'
+
+  /v1/namespaces/{namespace}:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/GetNamespaceResponse'
+        404:
+          description: Not Found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NoSuchNamespaceExample:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a namespace exists
+      operationId: namespaceExists
+      description: Check if a namespace exists.
+      responses:
+        '200':
+          description: Namesapce exists
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
```

Comment:
In the future, we may want to add a cascade option.

Reply:
Yeah. I hesitated to add this in (especially as I've been working on adding cascade support), but it is on the interface definition (as indicated by the `@throws` section of the javadoc): https://github.com/apache/iceberg/blob/c5d3d1ca19105d7f525f9912bd7f7f2e4507541b/api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java#L98-L105

Reply:
Yeah, we wanted to avoid boolean parameters on the catalog API where we could. The problem here is that it would be really inefficient to use the rest of this API to do a cascading drop because the client would need potentially hundreds of requests to drop each table and namespace individually, and the requests have relatively high latency. We may want to add a cascade flag to the catalog API as well, or a new method (`dropNamespaceCascade`).